### PR TITLE
Test framework - warn if global RouteBuilder may be wrong

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/4381

Adds warning into the log + test verifying it.

### CODE REVIEW

File: test-framework/junit5-extension-tests/src/test/java/org/apache/camel/quarkus/test/extensions/routeBuilder/RouteBuilderFailureTest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import io.quarkus.test.ContinuousTestingTestUtils;
+import io.quarkus.test.QuarkusDevModeTest;
+import org.apache.camel.quarkus.test.extensions.continousDev.HelloResource;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+/**
+ * Scenario when useRoutBuilder is FALSE and NO RouteBuilder is produced -> should fail.
```
Comment (aldettinger) [inline test-framework/junit5-extension-tests/src/test/java/org/apache/camel/quarkus/test/extensions/routeBuilder/RouteBuilderFailureTest.java:27]: small typo time permitting

File: test-framework/junit5-extension-tests/src/test/java/org/apache/camel/quarkus/test/extensions/routeBuilder/RouteBuilderWarningWithProducedBuilderTest.java [hunk 2]
```diff
@@ -16,46 +16,36 @@
...
-import org.jboss.shrinkwrap.api.asset.StringAsset;
-import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.awaitility.Awaitility;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
-public class RouteBuilderTest {
+/**
+ * Scenario when useRoutBuilder is TRUE and RouteBuilder is created via HelloRouteBuilder -> should succeed with
```
Comment (aldettinger) [inline test-framework/junit5-extension-tests/src/test/java/org/apache/camel/quarkus/test/extensions/routeBuilder/RouteBuilderWarningWithProducedBuilderTest.java:30]: Typo, time permitting

File: test-framework/junit5-extension-tests/src/test/java/org/apache/camel/quarkus/test/extensions/routeBuilder/RouteBuilderFailureTest.java (hunk 1 above)
Comment (JiriOndrusek) [reply test-framework/junit5-extension-tests/src/test/java/org/apache/camel/quarkus/test/extensions/routeBuilder/RouteBuilderFailureTest.java]: fixed

File: test-framework/junit5-extension-tests/src/test/java/org/apache/camel/quarkus/test/extensions/routeBuilder/RouteBuilderWarningWithProducedBuilderTest.java (hunk 2 above)
Comment (JiriOndrusek) [reply test-framework/junit5-extension-tests/src/test/java/org/apache/camel/quarkus/test/extensions/routeBuilder/RouteBuilderWarningWithProducedBuilderTest.java]: fixed

**Review by ppalaga — APPROVED** [review]

**Pull request merged**